src/main/scala/beam/utils/google_routes_db/GoogleRoutesDB.scala

```diff
@@ -0,0 +1,120 @@
+package beam.utils.google_routes_db
+
+import java.sql.{Connection, Statement}
+import java.time.{Instant, LocalDateTime}
+
+import beam.agentsim.events.PathTraversalEvent
+import beam.agentsim.events.handling.GoogleTravelTimeEstimationEntry
+import beam.sim.common.GeoUtils
+import beam.utils.FileUtils.using
+import beam.utils.google_routes_db.sql.{PSMapping, Update}
+import beam.utils.mapsapi.googleapi.GoogleRoutesResponse
+import com.google.maps.model.DirectionsRoute
+import com.typesafe.scalalogging.LazyLogging
+
+import scala.collection.mutable
+
+object GoogleRoutesDB extends LazyLogging {
+
+  //
+  // Queries
+  //
+
+  def queryGoogleRoutes(
+    ptes: TraversableOnce[PathTraversalEvent],
+    departureDateTime: LocalDateTime,
+    geoUtils: GeoUtils,
+    con: Connection
+  ): Map[PathTraversalEvent, Seq[GoogleTravelTimeEstimationEntry]] = {
+    using(con.prepareStatement(sql.Select.QueryGoogleRoutes.sql)) { ps =>
+      val result = mutable.Map.empty[PathTraversalEvent, Seq[GoogleTravelTimeEstimationEntry]]
+
+      ptes.foreach { pte =>
+        import sql.Select.QueryGoogleRoutes._
+
+        val arg = makeArg(pte, departureDateTime)
+        argPsMapping.mapPrepared(arg, ps)
+        val items = using(ps.executeQuery()) { rs => readItems(rs) }
```

Comment:
Should we batch these queries? It should be profiled on being hit by multiple PTE queries versus sending as a single transaction.